src/state_machine.zig

```diff
@@ -437,7 +436,9 @@ pub const StateMachine = struct {
     }
 
     fn commit_transfer_rollback(self: *StateMachine, c: Commit) void {
-        var t = self.get_transfer(c.id).?;
+        //TODO Ensure the commit exists before changing balances? Otherwise I can do a double commit rollback
+        var c_read = self.get_commit(c.id).?;
```

Comment:
Good idea. This could be an `assert` to avoid introducing another local variable name and because that fits what we're trying to do here: ensure something.

Reply:
Changed to an an `assert`.